qiskit_ode/signals/signals.py

```diff
@@ -327,202 +407,570 @@ def add_samples(self, start_sample: int, samples: List):
 
         self._samples = np.append(self._samples, samples)
 
+    def __str__(self) -> str:
+        """Return string representation."""
+        if self.name is not None:
+            return str(self.name)
 
-def signal_multiply(
-    sig1: Union[BaseSignal, float, int, complex], sig2: Union[BaseSignal, float, int, complex]
-) -> Signal:
-    r"""Implements mathematical multiplication between two signals.
-    Since a signal is represented by
-
-    .. math::
-
-        \Omega(t)*exp(2\pi i \nu t)
+        return "DiscreteSignal(dt={dt}, carrier_freq={freq}, phase={phase})".format(
+            dt=self.dt, freq=str(self.carrier_freq), phase=str(self.phase)
+        )
 
-    multiplication of two signals implements
 
-    .. math::
+class SignalSum(Signal):
+    """Class representing a sum of ``Signal`` objects."""
```

Comment:
I think this requires more details. Perhaps a few words on its use and how it is implemented.

Reply:
Updated.